src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c

```diff
@@ -1267,21 +1290,50 @@ static void icap_clean_bitstream_axlf(struct platform_device *pdev)
 	icap_clean_axlf_section(icap, PARTITION_METADATA);
 }
 
-static uint32_t convert_mem_type(const char *name)
+enum MEM_TAG convert_mem_tag(const char *name)
```

Comment:
This function is shared by all the sub device drivers. Please move it to xocl_xclbin.c